engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -27,37 +27,49 @@
     private static final Logger logger = LoggerFactory.getLogger(RenderTaskListGenerator.class);
 
     public RenderTaskListGenerator() {
-
     }
 
     public List<RenderPipelineTask> generateFrom(List<Node> orderedNodes) {
         logger.info("Task List: "); // TODO: remove in the future or turn it into debug
-        List<RenderPipelineTask> taskList = Lists.newArrayList();
-        for (Node node : orderedNodes) {
+        List<Object> intermediateList = generateIntermediateList(orderedNodes);
+        eliminateRedundantStateChanges(intermediateList);
```

Comment:
Hmmm... no, I wouldn't do it here. We don't want the intermediateList to loose information. We want the taskList to be as lean as possible. So, I think the elimination of redundant StateChanges only occurs by not having the associated tasks in the taskLists. In practice, as you iterate over the intermediateList to generate the taskList, I believe you can pick and choose the tasks to create. This should also help us when settings or the topology of the render graph changes. So, I'd simply remove the  `eliminateRedundantStateChanges(intermediateList);` call and do everything within `generateTaskList(intermediateList);`